src/main/scala/hercules/api/services/StatusService.scala

```diff
@@ -67,32 +72,20 @@ trait StatusService extends HerculesService {
               SendToAll(
                 "/user/master/active",
                 RequestMasterState())
-            )
+            )(timeout)
+
           /**
            * Take the response from master and map it to a StatusCode
            */
-          val response = request.map {
-            case MasterState(messagesNotYetProcessed, messagesInProcessing, failedMessages) => {
-              // @TODO Let some json marshaller handle the response instead
-              val status =
-                "messagesNotYetProcessed: {" +
-                  messagesNotYetProcessed.mkString(",") +
-                  "}, messagesInProcessing: {" +
-                  messagesInProcessing.mkString(",") +
-                  "} ,failedMessages: {" +
-                  failedMessages.mkString("}")
-              OK
-            }
-            /**
-             * Handle exceptions that may be thrown
-             */
-          }.recover {
-            case e: Exception =>
-              InternalServerError
+
+          import hercules.actors.masters.state.MasterStateJsonProtocol._
+
+          onComplete(request) {
+            case Success(state) => complete(state.asInstanceOf[MasterState])
+            case Failure(_)     => complete(InternalServerError)
           }
```

Comment:
nice refactoring, however, isn't there a case somewhere in there that isn't caught with onComplete? I remember something like either if an exception is thrown or if it times out